scripts/maintenance/migrate_project/src/cli.py

```diff
@@ -0,0 +1,78 @@
+from pathlib import Path
+
+import typer
+from db import (
+    db_connection,
+    get_project_and_files_to_migrate,
+    insert_file_meta_data,
+    insert_projects,
+)
+from models import Settings
+from r_clone import assemble_config_file, sync_file
+
+
+def main(config: Path = typer.Option(..., exists=True)):
+    assert config.exists()  # nosec
+    settings = Settings.load_from_file(config)
+    typer.echo(f"Detected settings:\n{settings.json(indent=2)}\n")
+
+    r_clone_config_path = assemble_config_file(
+        # destination is AWS S3
+        aws_access_key=settings.destination.s3.access_key,
+        aws_secret_key=settings.destination.s3.secret_key,
+        # source is MINIO
+        minio_access_key=settings.source.s3.access_key,
+        minio_secret_key=settings.source.s3.secret_key,
+        minio_endpoint=settings.source.s3.endpoint,
+    )
+    typer.echo(f"Rclone config:\n{r_clone_config_path.read_text()}\n")
+
+    with db_connection(settings.source.db) as src_db_conn, db_connection(
```

Comment:
nice, but would that not be much faster if it was async? you could gather. Maybe for the next iteration.